### PR TITLE
Adopt Divvi Up branch model

### PR DESCRIPTION
**Pull request by tgeoghegan**
Once this is merged, I still need to:

- [ ] branch `release/dap-draft-03` from `main`
- [ ] amend branch protection rules for `main` and `release/*` to require review from codeowners

### CODE REVIEW

**Comment by jbr** [comment]
How does this interact with the fact that VDAF and DAP are versioned distinctly but currently are synchronized in practice?

**Comment by tgeoghegan** [comment]
> How does this interact with the fact that VDAF and DAP are versioned distinctly but currently are synchronized in practice?

Specifying a draft of DAP suffices, because it, in turn, specifies a draft of VDAF. See, for example, the [normative references section on draft-ietf-ppm-dap-03](https://datatracker.ietf.org/doc/html/draft-ietf-ppm-dap-03#name-normative-references), which unambiguously has:

>[VDAF]
>Barnes, R., Patton, C., and P. Schoppmann, "Verifiable Distributed Aggregation Functions", Work in Progress, Internet-Draft, draft-irtf-cfrg-vdaf-03, 24 August 2022, <https://datatracker.ietf.org/doc/html/draft-irtf-cfrg-vdaf-03>.

**Comment by tgeoghegan** [comment]
I'm going to downgrade to draft while I debug the Actions changes, and will request reviews when it's ready.

**Comment by tgeoghegan** [comment]
This appears to be working the way I want it to, though we won't know for sure until something gets merged to `main` and we can see how the `docker push` actions play out.

File: .github/workflows/docker.yml [hunk 1]
```diff
@@ -19,14 +20,27 @@ jobs:
...
       id: get_commit
       run: echo "COMMIT=$(git rev-parse --short=7 HEAD)" >> $GITHUB_OUTPUT
+    - name: Get DAP draft
+      id: get_dap_draft
+      run: |
+        echo -n "DAP_DRAFT=" >> $GITHUB_OUTPUT
+        if [[ ${{ github.ref_name }} == "main" ]]; then
+          echo dap-draft-04 >> $GITHUB_OUTPUT
+        else
+          echo "${GITHUB_REF##*/}" >> $GITHUB_OUTPUT
```
Comment (divergentdave) [inline .github/workflows/docker.yml:30]: FYI, this part is going to run with GITHUB_REF set to pull request merge branches, i.e. `refs/pull/198/merge`, and thus it'll tag some images (never to be pushed) as `divviup_ts_interop_client:merge`.

**Review by divergentdave — APPROVED** [review]

File: .github/workflows/docker.yml (hunk 1 above)
Comment (tgeoghegan) [reply .github/workflows/docker.yml:30]: Yes, I know. That seems silly, but we have to put _something_, and it's easier to use the nonsense string `merge` than to try to figure out what `dap-draft-0x` it should be.

**Comment by tgeoghegan** [comment]
The right containers seem to have been published to our public and private container registries, so I think this worked out!

**Pull request merged**